installer/scripts/service_control

```diff
@@ -20,29 +23,75 @@ VAR_DIR=/var/opt/microsoft/omsagent
 ETC_DIR=/etc/opt/microsoft/omsagent
 BIN_DIR=/opt/microsoft/omsagent/bin
 
-WORKSPACE_REGEX='^[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}$'
+REGEX_UUID='[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}'
+WORKSPACE_REGEX="^$REGEX_UUID$"
+VALIDATION_SUPPLEMENT_MESSAGE=
+VALIDATION_SEVERITY=WARNING
 
 # Space seperated list of non oms workspaces
 NON_OMS_WS="scom"
 
-setup_variables()
-{
+# Begin Lower Level Utilities
+
+validate_file_has_line_pattern() {
+	local _fSpec=$1
+	local _pattA="$2"
+	local _boolOnly=$3
+	cat $_fSpec | grep -Eq "^$_pattA$"
+	if [ $? -eq 0 ]; then
+		return 0
+	fi
+	if [ -z "$_boolOnly" ]; then
+		echo "$VALIDATION_SEVERITY:  $_fSpec missing required pattern '$_pattA'."
+        echo $VALIDATION_SUPPLEMENT_MESSAGE
+	fi
+	exit 1
+}
+
+# End Lower Level Utilities
+
+# Begin FailCount Singleton Interface
+
+fcso_any_failures() {
+    return $FCSO_FailCount
+}
+
+fcso_increment() {
+    if [ $# -ne 3 ]; then
+        echo "FATAL:  Programmer error; 3 arguments required for fcso_increment!"
+        echo "$# arguments seen."
+        exit 99
```

Comment:
In other files, this is an "INTERNAL_ERROR" case for us, and we have a specific exit code defined at the top of the file for it ([example](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/bundle/bundle_skel.sh#L58)). If you want to exit with a code other than 1, please define that number at the top of the file for clarity.

Reply:
Also, you may want to mimic the message of the other example as well: `echo "INTERNAL ERROR: Incorrect number of parameters passed to fcso_increment" >&2`

Reply:
Good.  I'll follow that.  Sorry about the flub first time.  
BTW I was gonna suggest generally we move to logs always to stderr, and that is specifically an interesting question in omsadmin.sh.  Let's digress on that soon.